spark/AerospikeSparkPython.ipynb

```diff
@@ -0,0 +1,855 @@
+    "findspark.init()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pyspark\n",
+    "from pyspark.context import SparkContext\n",
+    "from pyspark.sql.context import SQLContext\n",
+    "from pyspark.sql.session import SparkSession\n",
+    "from pyspark.sql.types import StringType, StructField, StructType, ArrayType, IntegerType, MapType, LongType"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "AS_HOST =\"127.0.0.1:3000\"\n",
+    "namespace= \"test\" "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set up spark and point aerospike db to AS_HOST"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc = SparkContext.getOrCreate()\n",
+    "spark = SparkSession(sc)\n",
+    "sqlContext = SQLContext(sc)\n",
+    "spark.conf.set(\"aerospike.namespace\", namespace)\n",
+    "spark.conf.set(\"aerospike.seedhost\", AS_HOST)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create sample data and write to Aerospike database"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import random\n",
+    "num_records=1000\n",
+    "\n",
+    "schema = StructType( \n",
+    "    [\n",
+    "        StructField(\"_id\", IntegerType(), True),\n",
+    "        StructField(\"name\", StringType(), True),\n",
+    "        StructField(\"age\", IntegerType(), True),\n",
+    "        StructField(\"salary\",IntegerType(), True)\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "\n",
+    "inputBuf = []\n",
+    "for  i in range(1, num_records) :\n",
+    "         name = \"name\"  + str(i)\n",
+    "         age = i%100\n",
+    "         salary = 50000 + random.randint(1,50000)\n",
+    "         id_ = i \n",
+    "         inputBuf.append((id_, name, age,salary))\n",
+    "    \n",
+    "inputRDD = spark.sparkContext.parallelize(inputBuf)\n",
+    "inputDF=spark.createDataFrame(inputRDD,schema)\n",
+    "\n",
+    "\n",
+    "#Write the Sample Data to Aerospike\n",
+    "inputDF \\\n",
+    ".write \\\n",
+    ".mode('overwrite') \\\n",
+    ".format(\"com.aerospike.spark.sql\")  \\\n",
+    ".option(\"aerospike.seedhost\", AS_HOST) \\\n",
+    ".option(\"aerospike.namespace\", namespace) \\\n",
+    ".option(\"aerospike.writeset\", \"py_input_data\")\\\n",
+    ".option(\"aerospike.updateByKey\", \"_id\") \\\n",
+    ".save()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Schema in the Spark Connector\n",
+    "\n",
+    "-  Aerospike is schemaless, however spark adher to schema. After schema is decided upon (either through inference or given), data within the bins must honor the types. \n",
```

Comment:
...however spark adheres to a schema. After the schema...

Reply:
done.